File: app/src/main/java/com/rr/razasypelajes/Horses/Padres.java
package com.rr.razasypelajes.Horses;

import android.support.annotation.NonNull;

import com.rr.razasypelajes.Helpers.JSONBuildable;

import org.json.JSONException;
import org.json.JSONObject;

public class Padres extends JSONBuildable {
    private String img;
    private String cruza;

    public Padres(JSONObject json) {
        super(json);
    }

    public String getImg() {
        return img;
    }

    public String getCruza() {
        return cruza;
    }

    @NonNull
    @Override
    public String toString() {
        return "Padres{" +
                "img='" + img + '\'' +
                ", cruza='" + cruza + '\'' +
                '}';
    }

    @Override
    protected void fromJSON(JSONObject json) throws JSONException {
        this.img = json.getString("image");
        this.cruza = json.getString("cruza");
    }
}

File: app/src/main/java/com/rr/razasypelajes/horses/Padres.kt
package com.rr.razasypelajes.horses

import android.content.Context
import com.rr.razasypelajes.helpers.JSONBuildable
import com.rr.razasypelajes.helpers.JSONHelper
import org.json.JSONArray

import org.json.JSONException
import org.json.JSONObject

class Padres(json: JSONObject) : JSONBuildable(json) {
    var name: String? = null
        private set
    var img: String? = null
        private set
    var parents: List<String>? = null
        private set

    @Throws(JSONException::class)
    override fun fromJSON(json: JSONObject) {
        this.name = json.getString("name")
        this.img = json.getString("img")
        this.parents = JSONHelper.convertJsonArray(json.getJSONArray("parents"))
    }

    override fun toString(): String {
        return "Padres(name=$name, img=$img, parents=$parents)"
    }

    fun getChildImageId(context: Context): Int {
        return context.resources.getIdentifier(img, "drawable", context.packageName)
    }

    fun getParentsImagesIds(context: Context): List<Int> {
        var parentsImagesIds : List<Int> = ArrayList()
        for (parent in parents as List<String>){
            parentsImagesIds += context.resources.getIdentifier(parent, "drawable", context.packageName)
        }
        return parentsImagesIds
    }
}
